### ISSUE TITLE
Calling destroy on a model using other parameter than id wipes out all relations

### ISSUE THREAD

**Issue by paquettea** [balderdashy/waterline#799]
I have a model called Taxonmy with a custom ID. It has the following attributes

````javascript
attributes: {
	id : { type: 'string',primaryKey:true, required: true,unique:true },
	runningImport: { type: 'boolean',defaultsTo:false,internal:true,protected:true},
	term: { type: 'string', required: true },
	icon: { type: 'string'},
	requireSubscription: { type: 'boolean'}, 
	synonyms: { type: 'string'},
	description: { type: 'string'}, 
	builtBy : { type: 'string'},
	journalEntries : {
		collection : 'journalEntry',
		via : 'taxonomy'
	},
	parent: { model: 'Taxonomy'}, 
	children: {
		collection:'Taxonomy',
		via : 'parent'
	}
}
```

Calling the following
``` javascript
Taxonomy.destroy({runningImport:false}) //wipes all relations to journalEntries
```
wipes out all relations in the journalEntries many-to-many relations table (```journalentry_taxonomy__taxonomy_journalentries```) even for unaffected records

The only way I found for the relations to be maintanted is to delete record by record, by providing a specific ID on destroy (even fails with an array of IDS). The following works properly.

```javascript
Taxonomy.destroy({id:'expositions'})  //this works as expected
```

This seems unsuitable for many situation where you are destroying entities based on something else than it's id. It also fails if you provide a second criteria to the query. ex.
```javascript
Taxonomy.destroy({id:'expositions',runningImport:false})  //wipes out all relations 
```

**Comment by particlebanana** [comment]
Yikes!

**Comment by luislobo** [comment]
We have found that trying to remove any Object that has a many-to-many relationship, with an invalid ID, like '123', destroys all objects in the relationship. This is a HUGE bug. It has wiped all relationships in some of our dbs.

**Comment by devinivy** [comment]
Which adapter(s) are you using?  Let's try to patch this together ASAP!  I am concerned about waterline-sequel potentially allowing destroys of whole tables by virtue of having an empty where clause: https://github.com/balderdashy/waterline-sequel/blob/b53d72df60eb9d9b060c56fc3cc272bf7b8578ca/sequel/index.js#L194

**Comment by paquettea** [comment]
it was with mongo adapter.

**Comment by particlebanana** [comment]
Should be fixed in `0.10.17` thanks to https://github.com/balderdashy/waterline/pull/815 but let me know if you still have issues with custom primary keys.

@devinivy we can look at adding some checks in https://github.com/balderdashy/waterline-sequel/blob/master/sequel/where.js so that you never get an empty where clause.
